erpnext/accounts/doctype/sales_invoice/sales_invoice_patch.py

```
@@ -0,0 +1,72 @@
+# Patch for Sales Invoice to auto-close Sales Orders when line item discounts are applied
+# This patch addresses issue #49130
+
+# Method to add to SalesInvoice class:
+
+def _close_linked_sales_orders_on_discount(self):
+	"""Close linked Sales Orders when any discount is applied and the order is fully billed/delivered."""
+	
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)